src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -55,6 +55,8 @@ public class GitChangeSet extends ChangeLogSet.Entry {
             + PREFIX_COMMITTER + IDENTITY + "$");
     private static final Pattern RENAME_SPLIT = Pattern.compile("^(.*?)\t(.*)$");
 
+    private static final Pattern GITHUB_PRIVATE_EMAIL = Pattern.compile("^[0-9]+\\+(.*)@users\\.noreply\\.github\\.com$");
```

Comment:
I think the username portion needs to include at least one character, so I switched from `*` to `+` in the parenthesized expression.  I'd really like that change included in the pull request.

I find it easier to read exact character matches with special characters when the exact character is surrounded by square brackets.  That change is entirely optional.

```suggestion
    private static final Pattern GITHUB_PRIVATE_EMAIL = Pattern.compile("^[0-9]+[+](.+)@users[.]noreply[.]github[.]com$");
```

Can you confirm that it is intentional that this code is not handling the case of  username@users.noreply..github.com?

There are many Jenkins core commits with mail addresses like:

```text
dwnusbaum@users.noreply.github.com
escoem@users.noreply.github.com
farodin91@users.noreply.github.com
francoiscampbell@users.noreply.github.com
Freddie-V4@users.noreply.github.com
gmjck@users.noreply.github.com
hangsu-ma@users.noreply.github.com
```

Reply:
What do you think about https://issues.jenkins.io/browse/JENKINS-70342 / https://javadoc.jenkins.io/hudson/model/User.CanonicalIdResolver.html / https://github.com/jenkinsci/git-plugin/pull/1383#issuecomment-1364119731?


> Can you confirm that it is intentional that this code is not handling the case of [username@users.noreply](mailto:username@users.noreply)..github.com?

It was intentional because it was already handled generically in https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/hudson/plugins/git/GitChangeSet.java#L502. But then I decided to put it before https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/hudson/plugins/git/GitChangeSet.java#L488 and... Ups! Yeah, it should be handled in the regular expression.

Reply:
I think that the canonical ID resolver looks like a much better place to map SCM data to user IDs.  I was not aware of that entry point.  Thanks for detecting it.

Reply:
OK. I will close this PR then.